### PR TITLE
Get ms alias from github identity

### PR DESCRIPTION
**Pull request by sima-zhu**
**Summary**

The PR is using opensource rest API call to get ms alias from github identity.

**Expect returns:**

1. 200 status.
Return ms alias directly when call the scripts.
2. error response.
Print the API call error response and exit 1.

**Testing locally**

### CODE REVIEW

File: eng/common/scripts/Get-MsAlias-FromGithub.ps1 [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  [string]$GithubName,
+
+  [Parameter(Mandatory = $false)]
+  [string]$ApiVersion = "2019-10-01",
+
+  [Parameter(Mandatory = $false)]
+  [string]$ContentType = "application/json",
+
+  [Parameter(Mandatory = $false)]
+  [hashtable]$AdditionalHeaders
```
Comment (weshaggard) [inline eng/common/scripts/Get-MsAlias-FromGithub.ps1:28]: Do we we have a need to pass extra headers?

File: eng/common/scripts/Get-MsAlias-FromGithub.ps1 [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  [string]$AadToken,
+
+  [Parameter(Mandatory = $true)]
+  [string]$GithubName,
+
+  [Parameter(Mandatory = $false)]
+  [string]$ApiVersion = "2019-10-01",
+
+  [Parameter(Mandatory = $false)]
+  [string]$ContentType = "application/json",
```
Comment (weshaggard) [inline eng/common/scripts/Get-MsAlias-FromGithub.ps1:25]: Why do we need to pass in the apiversion or contenttype as parameters?

File: eng/common/scripts/Get-MsAlias-FromGithub.ps1 [hunk 3]
```diff
@@ -0,0 +1,64 @@
+<#
```
Comment (weshaggard) [inline eng/common/scripts/Get-MsAlias-FromGithub.ps1:1]: Lets name this something like:
`Get-AADIdentityFromGithubUser`

File: eng/common/scripts/Get-MsAlias-FromGithub.ps1 [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+.PARAMETER AdditionalHeaders
+Additional parameters for http request headers in key-value pair format, e.g. @{ key1 = val1; key2 = val2; key3 = val3}
+#>
+[CmdletBinding(SupportsShouldProcess = $true)]
+param(
+  [Parameter(Mandatory = $true)]
+  [string]$AadToken,
+
+  [Parameter(Mandatory = $true)]
+  [string]$GithubName,
```
Comment (weshaggard) [inline eng/common/scripts/Get-MsAlias-FromGithub.ps1:19]:
```suggestion
  [string]$GithubUser,
```

File: eng/common/scripts/Get-MsAlias-FromGithub.ps1 [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+Github identity. E.g sima-zhu
+.PARAMETER ContentType
+Content type of http requests.
+.PARAMETER AdditionalHeaders
+Additional parameters for http request headers in key-value pair format, e.g. @{ key1 = val1; key2 = val2; key3 = val3}
+#>
+[CmdletBinding(SupportsShouldProcess = $true)]
+param(
+  [Parameter(Mandatory = $true)]
+  [string]$AadToken,
```
Comment (weshaggard) [inline eng/common/scripts/Get-MsAlias-FromGithub.ps1:16]:
```suggestion
  [string]$opsAuthToken,
```

File: eng/common/scripts/Get-MsAlias-FromGithub.ps1 [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+    if ($AdditionalHeaders) {
+        return $Headers + $AdditionalHeaders
+    }
+    return $Headers
+}
+
+
+try {
+    $headers = Load-RequestHeaders
+    Write-Host "Fetching ms alias for github identity: $GithubName."
```
Comment (weshaggard) [inline eng/common/scripts/Get-MsAlias-FromGithub.ps1:50]:
```suggestion
    Write-Host "Fetching aad identity for github user: $GithubName."
```

File: eng/common/scripts/Get-MsAlias-FromGithub.ps1 [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+    exit 1
+}
+
+$resp | Write-Verbose
+
+if ($resp.aad) {
+    return $resp.aad.alias
+}
+
+LogError "Failed to retrieve the ms alias from given github identity: $GithubName."
```
Comment (weshaggard) [inline eng/common/scripts/Get-MsAlias-FromGithub.ps1:64]:
```suggestion
LogError "Failed to retrieve the aad identity from given github user: $GithubName."
```
Should we exit 1 here as well, or is what is the expected way to handle the errors from this script?

**Comment by sima-zhu** [comment]
Combined this PR with https://github.com/Azure/azure-sdk-tools/pull/2581
Closed this one.

**Pull request closed without merging**